src/components/Filter/Rating.js

```diff
@@ -1,27 +1,49 @@
 import React from "react";
 import "./css/rating.css";
+import { useProduct } from "../../Context/Product-Context";
 const Rating = () => {
+  const { dispatch } = useProduct();
   return (
     <>
       <div className="filter-sidebar-category">
         <span>
           <b>Rating</b>
         </span>
         <div className="filter-items">
-          <input type="radio" name="radio" />
-          4stars & above
+          <input
```

Comment:
radio input state is not controlled by react.
need add  **checked={rating === 4}**  same for other radio inputs as well
(get rating value from useProduct() context)
that's why your clear filter will not clear selected rating

Reply:
okay i will make changes